### ISSUE TITLE
Add member reference support to Java indexer. [phab:D696]

### ISSUE THREAD

**Issue by kythe** [google/kythe#969]
_Created by **mazurak** at 2016-02-06 02:34:22_: 

Make the Java indexer handle member references, and add tests that they're handled properly. Because constructor references are written "Type::new" (even while the internal name for a constructor is always "<init>"), I added fake names for "new"s; there might be a better way to do this, but this was a straightforward one.

**Comment by kythe** [comment]
_**schroederc** required changes at 2016-02-08 19:00:01_

**Comment by kythe** [comment]
_**schroederc** wrote at 2016-02-08 19:00:02_:

Please add the following to `RELEASES.md`:
```
Notable additions:
 - Java indexer: add Java 8 member reference support
```

**Comment by kythe** [comment]
_**mazurak** wrote at 2016-02-08 23:15:42_:

Add note to RELEASES.md.

**Comment by kythe** [comment]
_**mazurak** wrote at 2016-02-09 01:05:06_:

Correct Git error.

**Comment by kythe** [comment]
_**schroederc** accepted this revision at 2016-02-09 16:20:29_

**Comment by kythe** [comment]
_**mazurak** closed via commit e2328464cd0123f9c13261d8f818e5d729ad14b3 at 2016-02-10 00:52:23_
